### PR TITLE
fix: fuzz test

### PR DESCRIPTION
**Pull request by crodriguezvega**
This should fix the error introduced [here](https://github.com/cosmos/ics23/pull/151).

### CODE REVIEW

File: go/fuzz_test.go [hunk 1]
```diff
@@ -79,11 +79,11 @@ func FuzzVerifyNonMembership(f *testing.F) {
...
 	// 2. Now run the fuzzer.
 	f.Fuzz(func(t *testing.T, inputJSON []byte) {
-		bv := new(BatchVectorData)
-		if err := json.Unmarshal(inputJSON, bv); err != nil {
+		var bv BatchVectorData
+		if err := json.Unmarshal(inputJSON, &bv); err != nil {
 			return
 		}
-		if bv.Ref == nil || bv.Ref.RootHash == nil {
+		if bv.Ref == nil || bv.Proof == nil || bv.Ref.RootHash == nil {
```
Comment (crodriguezvega) [inline go/fuzz_test.go:86]: The problem was that not checking if proof was `nil`.

**Review by colin-axner — APPROVED** [review]
ACK, not sure why the code cov is decreasing. Would also be nice to require passing go tests to merge prs

**Comment by elias-orijtech** [comment]
LGTM. CC @odeke-em

**Comment by odeke-em** [comment]
Thank you very much @crodriguezvega! Thank you for the tag @elias-orijtech

**Pull request merged**